contrib/ros/worlds/block_world.py

```diff
@@ -0,0 +1,180 @@
+                osp.join(model_dir, 'cafe_table/model.sdf'))
+            Gazebo.load_gazebo_model(
+                'block',
+                Pose(position=Point(x=0.5725, y=0.1265, z=0.90)),
+                osp.join(model_dir, 'block/model.urdf'))
+            block = Block(
+                name='block',
+                initial_pos=(0.5725, 0.1265, 0.90),
+                random_delta_range=0.15,
+                resource=osp.join(model_dir, 'block/model.urdf'))
+            self._blocks.append(block)
+        else:
+            # TODO(gh/8: Sawyer runtime support)
+            pass
+
+    def _gazebo_update_block_states(self, data):
+        model_states = data
+        model_names = model_states.name
+
+        for block in self._blocks:
+            block_idx = model_names.index(block.name)
+            block_pose = model_states.pose[block_idx]
+            block.position = block_pose.position
+            block.orientation = block_pose.orientation
+
+    def reset(self):
+        if self._simulated:
+            self._reset_sim()
+        else:
+            # TODO(gh/8: Sawyer runtime support)
+            pass
+
+    def _reset_sim(self):
+        """
+        reset the simulation
+        """
+        # Randomize start position of object
+        for block in self._blocks:
+            block_random_delta = np.zeros(2)
+            while np.linalg.norm(block_random_delta) < 0.1:
+                block_random_delta = np.random.uniform(
+                    -block.random_delta_range,
+                    block.random_delta_range,
+                    size=2)
+            Gazebo.set_model_pose(
+                block.name,
+                new_pose=Pose(
+                    position=Point(
+                        x=block.initial_pos.x + block_random_delta[0],
+                        y=block.initial_pos.y + block_random_delta[1],
+                        z=block.initial_pos.z)))
+
+    def _reset_real(self):
+        # TODO(gh/8: Sawyer runtime support)
+        pass
+
+    def terminate(self):
+        if self._simulated:
+            for block in self._blocks:
+                Gazebo.delete_gazebo_model(block.name)
+        else:
+            # TODO(gh/8: Sawyer runtime support)
+            pass
+
+    def get_observation(self):
+        blocks_pos = np.array([])
+        blocks_ori = np.array([])
+
+        for block in self._blocks:
+            pos = np.array(
+                [block.position.x, block.position.y, block.position.z])
+            ori = np.array([
+                block.orientation.x, block.orientation.y, block.orientation.z,
+                block.orientation.w
+            ])
+            blocks_pos = np.concatenate((blocks_pos, pos))
+            blocks_ori = np.concatenate((blocks_ori, ori))
+
+        achieved_goal = np.squeeze(blocks_pos)
+
+        obs = np.concatenate((blocks_pos, blocks_ori))
+
+        return {'obs': obs, 'achieved_goal': achieved_goal}
```

Comment:
Dictionary returns from Python methods are an antipattern. Python has multiple return values, so use them (they will appear as a tuple when returned to the caller). 

If you would also like it to look like a dict, you can use `namedtuple`